### ISSUE TITLE
Improved job queue

### ISSUE THREAD

**Issue by joshua-laughner** [TCCON/auto-priors-rs#7]
The v1 approach to the job queue is: each time the "run" command is given, it queries the database to get the list of jobs in order of submission and priority. It then works through that list until all the jobs are done before being allowed to query the DB again. This means that if a higher priority job is added while its running, that job won't be run until the whole previous list of lower priority jobs is done.

Instead, what we should do is get the higher priority job each time a worker is ready. A couple options:

1. Periodically check for pending jobs. If there are any, see if we can spawn a new worker. If so, take the next one on the queue (by submission & priority) and spawn the worker for it. Whenever a worker finishes, set its job to done and decrement the count of active workers.
2. Spawn the max number of workers right away, and whenever one is idle, have it look for the next job.

I think the first might actually be easier (and not just because that's what I've got started). Its advantages are (1) it should be easier to change the number of workers on the fly and (2) since assigning jobs to workers is done in the synchronous part of the code, we're less likely to run into race conditions. In fact, I could add a new job state (`spawning` or `assigned`) which means that job has been assigned to a worker but not yet begun. Or that could just be `running`.

**Comment by joshua-laughner** [comment]
This is implemented, however there is currently a weird bug preventing >1 job from starting at once.

**Comment by joshua-laughner** [comment]
Fixed by commit d24c37924aba7f671cbcfeb6953004c8b3208dcb. Problem was a weird combination of factors. The main one is that I was using `std::process::Command` instead of `tokio::process::Command` to run the ginput jobs. I think what happened is the long running ginput jobs meant that their threads never hit an `await` until they finished running ginput, so other threads never got polled. 

However, that can't have been it alone: if I forced the LUT regeneration to run, it never blocked the other threads even while using `std::process::Command`. In testing, I found that I could stop the regular ginput jobs from blocking the other threads if I removed *any* reference to the database in the spawned function. It wasn't enough to replace the pool connection instances with detached connections or a clone of the pool, I had to actually remove the database connection entirely. So my guess is that there was a weird interaction that holding a reference to the database in any way stopped the other (non-ginput job) threads from sending their wakeup messages to the runtime, so they never got a chance to run.

Fortunately, switching to `tokio::process::Command` has fixed the issue, so we can allow the spawned tasks to keep a database connection.
